forge-game/src/main/java/forge/game/card/CardProperty.java

```diff
@@ -605,6 +605,12 @@ public static boolean cardHasProperty(Card card, String property, Player sourceC
             if (!card.canBeAttached(source, null)) {
                 return false;
             }
+        } else if (property.startsWith("CanBeTargetedBy")) {
+            final String def = property.substring(15);
+            final SpellAbility targetingSA = AbilityUtils.getDefinedSpellAbilities(source, def, spellAbility).get(0);
+            if (!card.canBeTargetedBy(targetingSA)) {
```

Comment:
you need to use `sa.canTarget`
hopefully it works even if already has "full" targets